uniffi/src/common.rs

```diff
@@ -30,22 +34,30 @@ impl UniffiCustomTypeWrapper for JsonValue {
     }
 
     fn unwrap(obj: Self) -> Self::Wrapped {
-        obj.to_string()
+        serde_json::to_string(&obj).expect("unable serialize json value")
     }
 }
 
-pub struct OnResult<T> {
-    sender: Mutex<RefCell<Option<oneshot::Sender<Result<T>>>>>,
-}
+pub(crate) type OnResultReceiver<T> = oneshot::Receiver<Result<T>>;
```

Comment:
I think it will be more convinient to define wrapper over oneshot

```rust
struct OnResultReceiver(oneshot::Receiver<Result<T>>);

impl<T> OnResultReceiver<T> {

pub(crate) async fn get(self) -> Result<T> {
         self.0
             .await
             .kind(ErrorKind::InvalidState, "unable receive callback result")?
     }
}

let (cb, receiver) = OnDIDResolverResult::new();
let res = receiver.get().await;
}
```

Reply:
Done